core/src/main/scala/org/http4s/util/ProcessApp.scala

```diff
@@ -0,0 +1,40 @@
+package org.http4s.util
+
+import org.http4s.internal.compatibility._
+
+import scalaz.{-\/, \/-}
+import scalaz.concurrent.Task
+import scalaz.stream.Process
+import scalaz.stream.Process.eval_
+import scalaz.stream.{async, wye}
+
+trait ProcessApp {
+  def main(args: List[String]): Process[Task, Unit]
+
+  private[this] val shutdownRequested =
+    async.signalOf(false)
+
+  final val requestShutdown: Task[Unit] =
+    shutdownRequested.set(true)
+
+  final def main(args: Array[String]): Unit = {
+    val halt = async.signalOf(false)
```

Comment:
Oops.  Yeah, I pulled it out so shutdowns could be requested outside a SIGTERM, and didn't clean up this one.